7_deploy_osh/roles/suse-osh-setup-workers/tasks/main.yml

```diff
@@ -1,15 +1,7 @@
 ---
-
 # Better to have another ansible role for running code against all worker nodes
-
 - name: Subvolumes setup on control plane worker nodes
   include: subvolumes-setup.yml
   when: ('osh-deploy-workers' in group_names)
   tags:
     - preinstall
-
-#- name: Generate and read kubeconfig from one caasp node
-#  include: get-kubeconfig.yml
-#  run_once: true
-#  tags:
-#    - preinstall
```

Comment:
Logic is to generate kubeconfig using caasp-cli on worker node against master. Generated file requires some tweaking, especially idp related certs path and server name. Its a work-in-progress as did not have time to look further in this area. 
It will be useful to have kube config generated as manually copied config via velum UI expires after some time and need to be manually generated abd copied again.

Reply:
I agree on this, but the code was checked out once, so I am sure it's fine if we remove comments :) 

I am fine with adding this automation, but in this case, we should refactor this: https://github.com/SUSE-Cloud/socok8s/blob/master/7_deploy_osh/roles/suse-osh-deploy/tasks/kubectl-install.yml#L38-L64
The code I linked above is relying on automation tooling to get the credentials. Sadly it relies on environment.json.

I am fine with removing this whole block of code in `kubectl-install.yml` together with the addition of local fetching of the worker node.

However, it won't work in my environment, as worker nodes don't have floating IPs, so I can't contact them directly.
But I would say that this assumption is maybe wrong, as I'd need to contact the worker anyway for different cases (subvolumes).

Should we consider this case and put it in the requirements/reference architecture?